src/map_extras.cpp

```diff
@@ -798,32 +798,103 @@ static void mx_portal( map &m, const tripoint &abs_sub )
 
 static void mx_minefield( map &m, const tripoint &abs_sub )
 {
+    const oter_id &center = overmap_buffer.ter( abs_sub.x, abs_sub.y, abs_sub.z );
+    const oter_id &north = overmap_buffer.ter( abs_sub.x, abs_sub.y - 1, abs_sub.z );
+    const oter_id &south = overmap_buffer.ter( abs_sub.x, abs_sub.y + 1, abs_sub.z );
+    const oter_id &west = overmap_buffer.ter( abs_sub.x - 1, abs_sub.y, abs_sub.z );
+    const oter_id &east = overmap_buffer.ter( abs_sub.x + 1, abs_sub.y, abs_sub.z );
+
+    const bool bridge_at_center = is_ot_type( "bridge", center );
+    const bool bridge_at_north = is_ot_type( "bridge", north );
+    const bool bridge_at_south = is_ot_type( "bridge", south );
+    const bool bridge_at_west = is_ot_type( "bridge", west );
+    const bool bridge_at_east = is_ot_type( "bridge", east );
+
+    const bool road_at_center = is_ot_type( "road", center );
+    const bool road_at_north = is_ot_type( "road", north );
+    const bool road_at_south = is_ot_type( "road", south );
+    const bool road_at_west = is_ot_type( "road", west );
+    const bool road_at_east = is_ot_type( "road", east );
+
     const int num_mines = rng( 6, 20 );
+    const std::string text = _( "DANGER! MINEFIELD!" );
+    int x, y, x1, y1 = 0;
 
-    for( int i = 0; i < num_mines; i++ ) {
-        // No mines at the extreme edges: safe to walk on a sign tile
-        const int x = rng( 1, SEEX * 2 - 2 ), y = rng( 1, SEEY * 2 - 2 );
-        if( m.has_flag( "DIGGABLE", x, y ) ) {
-            mtrap_set( &m, x, y, tr_landmine_buried );
-        } else {
-            mtrap_set( &m, x, y, tr_landmine );
+    if( bridge_at_north && !bridge_at_center && road_at_south ) {
+
+        line_furn( &m, f_sandbag_half, 3, 4, 3, 7 );
+        line_furn( &m, f_sandbag_half, 3, 7, 9, 7 );
+        line_furn( &m, f_sandbag_half, 9, 4, 9, 7 );
+
+        for( const auto &loc : g->m.points_in_radius( { 6, 4, abs_sub.z }, 3, 0 ) ) {
+            if( one_in( 4 ) ) {
+                m.add_item_or_charges( loc, item( "762_51_casing" ) );
```

Comment:
The US military primarily uses 5.56 rounds, so perhaps this should be changed to casings of that variety?

Reply:
These are casings from humvee's m60.

Reply:
To the right there is another sandbag block with a dead soldier, so I replaced 7.62 with 5.56 there.